osquery/config/plugins/tls.cpp

```diff
@@ -46,32 +48,33 @@ CLI_FLAG(uint64,
          0,
          "Optional interval in seconds to re-read configuration");
 
+/// Option to always do enroll at startup time
+CLI_FLAG(bool,
+         config_tls_enroll_on_startup,
+         false,
+         "Optional setting to do enroll on startup");
+
 DECLARE_bool(tls_secret_always);
 DECLARE_string(tls_enroll_override);
 DECLARE_bool(tls_node_api);
 
-class TLSConfigPlugin;
-
-class TLSConfigPlugin : public ConfigPlugin,
-                        std::enable_shared_from_this<TLSConfigPlugin> {
- public:
-  Status setUp() override;
-  Status genConfig(std::map<std::string, std::string>& config) override;
-
- protected:
-  /// Calculate the URL once and cache the result.
-  std::string uri_;
-};
-
-class TLSConfigRefreshRunner : public InternalRunnable {
- public:
-  /// A simple wait/interruptible lock.
-  void start();
-};
-
 REGISTER(TLSConfigPlugin, "config", "tls");
 
 Status TLSConfigPlugin::setUp() {
+  if (FLAGS_config_tls_enroll_on_startup && !FLAGS_disable_enrollment) {
```

Comment:
I'd like to disconnect this "ephemeral" enroll concept from the specific TLS config and TLS enroll plugins if it's not too much work?